Show systemd container without privileged mode or SYS_ADMIN capabilit…

1efty opened the pull request:
I haven't been using privileged mode or SYS_ADMIN capabilities on my systemd-enabled containers.  It's good information knowing that you do not need extended privileged in a container to run systemd services.  

https://developers.redhat.com/blog/2016/09/13/running-systemd-in-a-non-privileged-container/

#### PR Type

- Docs Pull Request

decentral1se commented on doc/source/examples.rst:
```diff
@@ -98,16 +98,17 @@ Linux system or manage services for such things as start-up order, dependency
 
 To start a service which requires systemd, configure `molecule.yml` with a
-systemd compliant image, capabilities, volumes, and command as follows.
+systemd compliant image, tmpfs, volumes, and command as follows. [4]_
```
Are the people who want `capabilities: SYS_ADMIN` losing something now?

1efty replied:
Makes sense.  I added back the SYS_ADMIN example and made it clear that it is a privileged container.

I also removed the top paragraph since it seems superfluous now and the first reference since it's 404.  I also added some official references from docker explaining privileged mode and a reference to the capabilities man page further explaining the SYS_ADMIN capability.

decentral1se replied:
Thanks!

decentral1se commented:
> The new example use the base containers and does say nothing about the systemd ones.

@1efty, do you know if we still get the same functionality for centos:7 image vs. centos:systemd!? Do we also need to add a note that you may need to use those containers as well? What are the differences?

Sorry to drag you through the mud here but this is actually quite a key documentation example and I'd be really happy if we nail this down to cover the majority of needs. This is one of the major pain points for people running containers with their CI and enabling services ...

1efty commented:
@decentral1se I've been using the normal centos container for months now with no issues, starting, stopping, restarting systemd services such as apache, postgresql, varnish, etc.  

@pilou- made an addition which seems to explain why the base centos container works, and also adds an example showing how to make the official debian example work.

https://github.com/1efty/molecule/pull/1

decentral1se commented:
Ah, we seem to be missing the `--signoff` on that commit @pilou-  (see failing checks ...).

decentral1se commented:
> I would like to see a ticket describing the problem

It was in the title: "Show systemd container without privileged mode". Some people need that.

pilou- commented:
> Ah, we seem to be missing the --signoff on that commit @pilou- (see failing checks ...).

I made a PR on this PR, is there a way to fix that ?

decentral1se commented:
Someone needs to `commit --amend` that last commit and add `Signed-off-by: ...` unfortunately.

1efty commented:
Shoulda caught that.

@pilou- want to amend your commit and make a new pr?

decentral1se commented:
https://travis-ci.com/ansible/molecule/jobs/187880930 passed :+1:

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
-checking, and failed service recovery. [1]_
+To start a service which requires systemd, in an `unprivileged` container,
+configure `molecule.yml` with a systemd compliant image, tmpfs, volumes,
+and command as follows. [1]_
```
Hmmm, sorry to nitpick now but no other part of the documentation is using these `[N]` links ...

Would you mind fixing to use the typical RST links like:

```
`my link`_

.. _my link: URL
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
+          - /sys/fs/cgroup:/sys/fs/cgroup:ro
+
+Note that centos:7 image contains a `seccomp security profile for Docker`_ which allows to use Systemd.
+When needed, such security profile can be reused (for example the one available in Fedora [2]_:
```
```suggestion
When needed, such security profiles can be reused (for example the one available in Fedora) [2]_:
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
 
-The developer can also opt to start the container with extended privileges.
+The developer can also opt to start the container with extended privileges,
+by either giving it `SYS_ADMIN` capabilites or running it in `privileged` mode. [3]_
```
```suggestion
by either giving it ``SYS_ADMIN`` capabilites or running it in ``privileged`` mode.
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
 .. important::
 
-    Use caution when using `privileged` mode. [2]_ [3]_
+    Use caution when using `privileged` mode or `SYS_ADMIN`
```
```suggestion
    Use caution when using ``privileged`` mode or ``SYS_ADMIN``
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
+    capabilities as it grants the container elevated access to the
+    underlying system. [4]_ [5]_ [6]_
+
+To limit the scope of the extended privileges, grant `SYS_ADMIN`
```
```suggestion
To limit the scope of the extended privileges, grant ``SYS_ADMIN``
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
+    underlying system. [4]_ [5]_ [6]_
+
+To limit the scope of the extended privileges, grant `SYS_ADMIN`
+capabilities along with the same image, command, and volumes as shown in the `unprivileged` example.
```
```suggestion
capabilities along with the same image, command, and volumes as shown in the ``unprivileged`` example.
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
+        volumes:
+          - /sys/fs/cgroup:/sys/fs/cgroup:ro
+
+To start the container in `privileged` mode, set the privileged flag along with the
```
```suggestion
To start the container in ``privileged`` mode, set the privileged flag along with the
```

decentral1se commented on doc/source/examples.rst:
```diff
@@ -92,42 +92,80 @@ project.
+          - /sys/fs/cgroup:/sys/fs/cgroup:ro
+
+To start the container in `privileged` mode, set the privileged flag along with the
+same image and command as shown in the `unprivileged` example.
```
```suggestion
same image and command as shown in the ``unprivileged`` example.
```

1efty commented:
Updated the formatting on molecule.yml to match the rest of the document.  Used non-privileged instead of unprivileged since the reference uses that term.  Also updated all of the links.

decentral1se commented:
:rocket:
